### PR TITLE
supports s3 storage

### PR DESCRIPTION
**Pull request by stringang**
# Which chart

- mysqldump

# What this PR does / why we need it

- Supports S3 storage.

# Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

# Special notes for your reviewer

- .

# Checklist

- [x] [DCO](https://github.com/kokuwaio/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] README is up to date regarding breaking changes / upgrading instructions

### CODE REVIEW

File: charts/mysqldump/templates/configmap-script.yaml [hunk 1]
```diff
@@ -85,6 +85,13 @@ data:
     rcu=$?
     {{ end }}
 
+    {{ if .Values.upload.s3.enabled -}}
+    echo "upload files to s3 storage bucket {{ .Values.upload.s3.bucketname }}"
+    mc alias set s3 ${S3_ENDPOINT} ${S3_ACCESSKEY} ${S3_SECRET_KEY}
```
Comment (monotek) [inline charts/mysqldump/templates/configmap-script.yaml:90]: please use "aws s3 cp" comnmand.
i'll add it to the container.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/mysqldump/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
-version: 6.0.0
-appVersion: v3.0.16
+version: 6.1.0
+appVersion: v3.1.0
```
Comment (monotek) [inline charts/mysqldump/Chart.yaml:3]:
```suggestion
appVersion: v3.1.1
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/mysqldump/templates/configmap-script.yaml (hunk 1 above)
Comment (stringang) [reply charts/mysqldump/templates/configmap-script.yaml]: completed

File: charts/mysqldump/templates/configmap-script.yaml [hunk 3]
```diff
@@ -85,6 +85,13 @@ data:
     rcu=$?
     {{ end }}
 
+    {{ if .Values.upload.s3.enabled -}}
+    echo "upload files to s3 storage bucket {{ .Values.upload.s3.bucketname }}"
+    aws configure set aws_access_key_id "$S3_ACCESSKEY" && aws configure set aws_secret_access_key "$S3_SECRET_KEY" && aws configure set region "$S3_REGION"
+    find ${BACKUP_DIR} -type d -name 'lost+found' -prune -o -maxdepth 2 -name "${TIMESTAMP}_*.sql.gz" -type f -exec aws --endpoint-url ${S3_ENDPOINT} s3 cp {} s3://${S3_BACKETNAME} \;
```
Comment (monotek) [inline charts/mysqldump/templates/configmap-script.yaml:91]: Typo?
```suggestion
    find ${BACKUP_DIR} -type d -name 'lost+found' -prune -o -maxdepth 2 -name "${TIMESTAMP}_*.sql.gz" -type f -exec aws --endpoint-url ${S3_ENDPOINT} s3 cp {} s3://${S3_BUCKETNAME} \;
```

File: charts/mysqldump/templates/configmap-script.yaml [hunk 4]
```diff
@@ -85,6 +85,13 @@ data:
     rcu=$?
     {{ end }}
 
+    {{ if .Values.upload.s3.enabled -}}
+    echo "upload files to s3 storage bucket {{ .Values.upload.s3.bucketname }}"
+    aws configure set aws_access_key_id "$S3_ACCESSKEY" && aws configure set aws_secret_access_key "$S3_SECRET_KEY" && aws configure set region "$S3_REGION"
```
Comment (monotek) [inline charts/mysqldump/templates/configmap-script.yaml:90]: cant we use a credentials secret here, as for the other methods?

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/mysqldump/templates/configmap.yaml [hunk 5]
```diff
@@ -17,3 +17,8 @@ data:
   OS_USERNAME: {{ .Values.upload.openstack.user }}
   OS_USER_DOMAIN_NAME: {{ .Values.upload.openstack.userDomain }}
 {{- end }}
+{{- if .Values.upload.s3.enabled }}
+  S3_ENDPOINT: {{ .Values.upload.s3.endpoint }}
+  S3_ACCESSKEY: {{ .Values.upload.s3.accesskey }}
+  S3_BACKETNAME: {{ .Values.upload.s3.bucketname }}
```
[the remark this answers is not in the document]
Comment (stringang) [reply charts/mysqldump/templates/configmap.yaml]: completed

File: charts/mysqldump/templates/configmap-script.yaml (hunk 3 above)
Comment (stringang) [reply charts/mysqldump/templates/configmap-script.yaml]: completed

File: charts/mysqldump/templates/configmap-script.yaml [hunk 6]
```diff
@@ -85,6 +85,13 @@ data:
     rcu=$?
     {{ end }}
 
+    {{ if .Values.upload.s3.enabled -}}
+    echo "upload files to s3 storage bucket {{ .Values.upload.s3.bucketname }}"
+    mc alias set s3 ${S3_ENDPOINT} ${S3_ACCESSKEY} ${S3_SECRET_KEY}
+    find ${BACKUP_DIR} -type d -name 'lost+found' -prune -o -maxdepth 2 -name "${TIMESTAMP}_*.sql.gz" -type f -exec mc cp {} s3/${S3_BACKETNAME}/${TIMESTAMP}/ \;
```
[the remark this answers is not in the document]
Comment (stringang) [reply charts/mysqldump/templates/configmap-script.yaml]: completed

File: charts/mysqldump/templates/configmap-script.yaml (hunk 4 above)
Comment (stringang) [reply charts/mysqldump/templates/configmap-script.yaml:90]: If credentials are not used, the only way to upload with temporary credentials is through AWS STS.

**Review by monotek — CHANGES_REQUESTED** [review]
i've update the ci part but the installation fails when it tries to upgrade from version  6.0.0:

```
mysqldump-28173050-97s6x   0/1     CreateContainerConfigError   0          3m55s
```

describe pod says:

```
  Warning  Failed     8s (x4 over 33s)  kubelet            Error: secret "mysqldump-s3-secretkey" not found
```

**Comment by monotek** [comment]
Still fails:

```
Error: UPGRADE FAILED: template: mysqldump/templates/secret-s3.yaml:1:14: executing "mysqldump/templates/secret-s3.yaml" at <.Values.upload.s3.enabled>: nil pointer evaluating interface {}.enabled
```

File: charts/mysqldump/values.yaml [hunk 7]
```diff
@@ -140,6 +140,17 @@ upload:
...
+    bucketname: mysql-backup
+    endpoint: https://yourdomain/
+    region: "us-east-1"
+    accesskey: ""
+    secretkey: ""
+    # existingSecret can be enabled to use an existing secret
+    existingSecret: ""
+    # existingSecretKey defines the key to use, or 'openstack-backup-password' if not set
+    existingSecretKey: ""
 resources: {}
```
Comment (monotek) [inline charts/mysqldump/values.yaml:154]:
```suggestion
    existingSecretKey: ""

resources: {}
```

**Comment by stringang** [comment]
> Still fails:
> 
> ```
> Error: UPGRADE FAILED: template: mysqldump/templates/secret-s3.yaml:1:14: executing "mysqldump/templates/secret-s3.yaml" at <.Values.upload.s3.enabled>: nil pointer evaluating interface {}.enabled
> ```

It's still the same problem. I've reviewed the code, and there shouldn't be any issues.

**Pull request merged**